demos/security_barrier_camera_demo/main.cpp

```diff
@@ -804,26 +811,26 @@ int main(int argc, char* argv[]) {
         worker->join();
         const auto t1 = std::chrono::steady_clock::now();
 
-        for (auto& net : std::array<std::vector<InferRequest>, 3>{context.detectorsInfers.getActualInferRequests(),
-                context.attributesInfers.getActualInferRequests(), context.platesInfers.getActualInferRequests()}) {
-            for (InferRequest& ir : net) {
-                ir.Wait(IInferRequest::WaitMode::RESULT_READY);
-                if (FLAGS_pc) {
-                    printPerformanceCounts(ir, std::cout);
+        if (FLAGS_pc) {  // Show performace results
+            for (auto& net : std::array<std::pair<std::vector<InferRequest>, std::string>, 3>{
+                std::make_pair(context.detectorsInfers.getActualInferRequests(), FLAGS_d),
+                    std::make_pair(context.attributesInfers.getActualInferRequests(), FLAGS_d_va),
+                    std::make_pair(context.platesInfers.getActualInferRequests(), FLAGS_d_lpr)}) {
+                for (InferRequest& ir : net.first) {
+                    ir.Wait(IInferRequest::WaitMode::RESULT_READY);
+                    printPerformanceCounts(ir, std::cout, getFullDeviceName(mapDevices, net.second));
```

Comment:
Please, revert this change. It is essential to wait for the rest inference calls for the demo when finishing, even if we do not need to `printPerformanceCounts`. It can end up with segmentation fault otherwise.

Reply:
Yes, got it. Looks like incorrect merge. Fixed.